crates/nu-plugin/src/protocol/plugin_custom_value.rs

```diff
@@ -363,3 +478,25 @@ impl PluginCustomValue {
         }
     }
 }
+
+impl Drop for PluginCustomValue {
+    fn drop(&mut self) {
+        // If the custom value specifies notify_on_drop and this is the last copy, we need to let
+        // the plugin know about it if we can.
+        if self.source.is_some() && self.notify_on_drop() && Arc::strong_count(&self.shared) == 1 {
```

Comment:
As I understand it, if `Arc::strong_count()` returns 1, it can only ever return more than 1 if the current thread clones it. If there are more copies you have to carefully consider who has it and what that might mean, but if it's only 1, that's guaranteed to be exclusive.

Even the functions in `Arc` for getting a mutable reference or taking the inner value just check that the counters are 1, because there's no other way for them to increase (at least in safe Rust)